custom/icds_reports/utils.py

```diff
@@ -176,3 +184,1196 @@ def render_html(self):
         return render_to_string("icds_reports/partials/column.html", dict(
             col=column_params
         ))
+
+
+def percent_increase(prop, data, prev_data):
+    current = 0
+    previous = 0
+    if data:
+        current = data[0][prop]
+    if prev_data:
+        previous = prev_data[0][prop]
+    return ((current or 0) - (previous or 0)) / float(previous or 1) * 100
+
+
+def percent_diff(property, current_data, prev_data, all):
+    current = 0
+    curr_all = 1
+    prev = 0
+    prev_all = 1
+    if current_data:
+        current = (current_data[0][property] or 0)
+        curr_all = (current_data[0][all] or 1)
+
+    if prev_data:
+        prev = (prev_data[0][property] or 0)
+        prev_all = (prev_data[0][all] or 1)
+
+    current_percent = current / float(curr_all) * 100
+    prev_percent = prev / float(prev_all) * 100
+    return current_percent - prev_percent
+
+
+def get_value(data, prop):
+    return (data[0][prop] or 0) if data else 0
+
+
+def get_location_filter(location, domain, config):
+    loc_level = 'state'
+    if location:
+        try:
+            sql_location = SQLLocation.objects.get(location_id=location, domain=domain)
+            aggregation_level = sql_location.get_ancestors(include_self=True).count() + 1
+            location_code = sql_location.site_code
+            if sql_location.location_type.code != LocationTypes.AWC:
+                loc_level = LocationType.objects.filter(
+                    parent_type=sql_location.location_type,
+                    domain=domain
+                )[0].code
+            else:
+                loc_level = LocationTypes.AWC
+            location_key = '%s_site_code' % sql_location.location_type.code
+            config.update({
+                location_key: location_code,
+                'aggregation_level': aggregation_level
+            })
+        except SQLLocation.DoesNotExist:
+            pass
+    return loc_level
+
+
+def get_system_usage_data(yesterday, before_yesterday, config):
```

Comment:
Why pass in yesterday and before yesterday? you should be able to figure those out inside this method